### PR TITLE
CMake: Fix Build & Install Option Names

### PR DESCRIPTION
**Pull request by ax3l**
We should not overwrite `CMAKE_` variables but instead set our own. This corrects this.

This also fixes that the cmake install path was unset in superbuilds failing the configure steps unless set by the calling project did set `CMAKE_INSTALL_CMAKEDIR` (or turned `openPMD_INSTALL=OFF`):
```
CMake Error at openpmd-src/CMakeLists.txt:1080 (install):
  install EXPORT given no DESTINATION!

CMake Error at openpmd-src/CMakeLists.txt:1085 (install):
  install FILES given no DESTINATION!
```
seen with https://github.com/fnalacceleratormodeling/synergia2/pull/100

Follow-up to #1313

cc @DerNils-git @s-sajid-ali

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -43,22 +43,73 @@ endif()
...
 endif()
-if(NOT CMAKE_RUNTIME_OUTPUT_DIRECTORY)
-    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin")
+if(NOT openPMD_RUNTIME_OUTPUT_DIRECTORY)
+    if(CMAKE_RUNTIME_OUTPUT_DIRECTORY)
+        set(openPMD_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
+    else()
+        set(openPMD_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin")
+    endif()
 endif()
```
Comment (ax3l) [inline CMakeLists.txt:66]: FYI:
there is also
* [CMAKE_PDB_OUTPUT_DIRECTORY](https://cmake.org/cmake/help/latest/variable/CMAKE_PDB_OUTPUT_DIRECTORY.html#variable:CMAKE_PDB_OUTPUT_DIRECTORY)
* [CMAKE_COMPILE_PDB_OUTPUT_DIRECTORY](https://cmake.org/cmake/help/latest/variable/CMAKE_COMPILE_PDB_OUTPUT_DIRECTORY.html#variable:CMAKE_COMPILE_PDB_OUTPUT_DIRECTORY)

File: CMakeLists.txt [hunk 2]
```diff
@@ -857,6 +945,11 @@ if(openPMD_BUILD_TESTING)
     foreach(testname ${openPMD_TEST_NAMES})
         add_executable(${testname}Tests test/${testname}Test.cpp)
         openpmd_cxx_required(${testname}Tests)
+        set_target_properties(${testname}Tests PROPERTIES
+            RUNTIME_OUTPUT_DIRECTORY ${openPMD_RUNTIME_OUTPUT_DIRECTORY}
+            PDB_OUTPUT_DIRECTORY ${openPMD_RUNTIME_OUTPUT_DIRECTORY}
+            COMPILE_PDB_OUTPUT_DIRECTORY ${openPMD_RUNTIME_OUTPUT_DIRECTORY}
```
Comment (ax3l) [inline CMakeLists.txt:951]:
```suggestion
            COMPILE_PDB_OUTPUT_DIRECTORY ${openPMD_COMPILE_RUNTIME_OUTPUT_DIRECTORY}
```

**Review by nischild — APPROVED** [review]
I could not find any `set(...)` statements which could be mixed up with other variables in other projects besides `CMAKE_CXX_FLAGS` which is probably set to be used in subprojects.

**Review by s-sajid-ali — APPROVED** [review]

File: CMakeLists.txt [hunk 3]
```diff
@@ -43,22 +43,87 @@ endif()
...
+    if(CMAKE_COMPILE_PDB_OUTPUT_DIRECTORY)
+        set(openPMD_COMPILE_PDB_OUTPUT_DIRECTORY ${CMAKE_COMPILE_PDB_OUTPUT_DIRECTORY})
+    else()
+        set(openPMD_COMPILE_PDB_OUTPUT_DIRECTORY ${openPMD_LIBRARY_OUTPUT_DIRECTORY})
+    endif()
 endif()
+
 # install directories
-if(CMAKE_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR)
+if(NOT CMAKE_INSTALL_LIBDIR)
```
Comment (ax3l) [inline CMakeLists.txt]:
```suggestion
if(NOT CMAKE_INSTALL_LIBDIR AND NOT WIN32)
```

**Comment by ax3l** [comment]
@franzpoeschel all green now :)

**Pull request merged**